lib/harvester.js

```diff
@@ -170,6 +170,7 @@ Harvester.prototype.resource = function (name, schema, options) {
     console.warn('Warning: resource "' + name + '" was already defined.');
     return this;
   }
+  schema = this._convertToLegacySchema(schema);
```

Comment:
Looks like for now, you're not storing the new schema. You'll need to in order to do some of the fancy pants stuff coming down the line, and others who are building on your work will be unblocked by this being done centrally by you in this pullrq.